nucypher/blockchain/eth/actors.py

```diff
@@ -275,10 +292,17 @@ class Miner(NucypherTokenActor):
     class MinerError(NucypherTokenActor.ActorError):
         pass
 
-    def __init__(self, is_me: bool, start_staking_loop: bool = True, *args, **kwargs) -> None:
+    def __init__(self,
+                 is_me: bool,
+                 start_staking_loop: bool = True,
+                 economics: TokenEconomics = None,
+                 *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
         self.log = Logger("miner")
         self.is_me = is_me
+        if not economics:
+            economics = TokenEconomics()
+        self.economics = economics
```

Comment:
I see that economics used not only for deployers but for actors too. I think it will be useful to add second way of `TokenEconomics` initialization - by reading parameters from contract using agent.

Reply:
Opened: <https://github.com/nucypher/nucypher/issues/935>